packages/vector_graphics_compiler/lib/src/svg/clipping_optimizer.dart

```diff
@@ -0,0 +1,224 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:vector_graphics_compiler/src/svg/node.dart';
+import 'package:vector_graphics_compiler/src/svg/resolver.dart';
+import 'package:vector_graphics_compiler/src/svg/visitor.dart';
+import 'package:vector_graphics_compiler/src/svg/masking_optimizer.dart';
+import 'package:vector_graphics_compiler/vector_graphics_compiler.dart';
+import 'package:vector_graphics_compiler/src/svg/path_ops.dart' as path_ops;
+
+class _Result {
+  _Result(this.node);
+
+  final Node node;
+  int childCount = 0;
+  List<Node> children = <Node>[];
+  Node parent = Node.empty;
+  bool deleteClipNode = true;
+}
+
+/// Applies and removes trivial cases of clipping.
```

Comment:
Consider adding an additional paragraph that clarifies what counts as a trivial case

Reply:
fixed.